docs/core/deploying/single-file.md

```diff
@@ -11,6 +11,10 @@ Bundling all application-dependent files into a single binary provides an applic
 
 Single File deployment is available for both the [framework-dependent deployment model](index.md#publish-framework-dependent) and [self-contained applications](index.md#publish-self-contained). The size of the single file in a self-contained application will be large since it will include the runtime and the framework libraries. The single file deployment option can be combined with [ReadyToRun](ready-to-run.md) and [Trim (an experimental feature in .NET 5.0)](trim-self-contained.md) publish options.
 
+## Output differences from .NET 3.x
+
+In .NET Core 3.x, publishing as a single file produced exactly one file, with all files next to the app bundled and extracted at startup. In .NET 5+, all managed DLLs are loaded from memory and there is no extraction by default. On Windows, this means that the managed binaries are embedded in the single-file bundle, but the native binaries of the core runtime itself are seperate files. To embed those files for extraction and get exactly one output file, like in .NET Core 3.x, set the property `IncludeNativeLibrariesForSelfExtract` to `true`. For more information about extraction, see [Other considerations](#other-considerations).
```

Comment:
Let me use a suggestion to see if I understand it correctly, please correct.

```suggestion
In .NET Core 3.x, publishing as a single file produced exactly one file, consisting of the app itself, dependencies, and any other files in the folder during publish. When the app starts, the single file app was extracted to a temporary folder and run from there. Starting with .NET 5, only managed DLLs are bundled with the app into a single executable. When the app starts, the managed DLLs are extracted and loaded in memory, avoiding the extraction to a temporary folder. On Windows, this means that the managed binaries are embedded in the single-file bundle, but the native binaries of the core runtime itself are separate files. To embed those files for extraction and get exactly one output file, like in .NET Core 3.x, set the property `IncludeNativeLibrariesForSelfExtract` to `true`. For more information about extraction, see [Other considerations](#other-considerations).
```

This does call out Windows, what is it for other platforms?

Reply:
This looks right. On Linux the runtime itself is in the single-file, so the only extra files are 3rd party native binaries from NuGet or similar.

Reply:
what about external file resources, like json, txt, jpg, etc?

Reply:
Run from the folder that invoked it -- external assets aren't included at all, they're left by the app.